DynamicData/Cache/Internal/TreeBuilder.cs

```diff
@@ -13,19 +14,23 @@ internal class TreeBuilder<TObject, TKey>
     {
         private readonly IObservable<IChangeSet<TObject, TKey>> _source;
         private readonly Func<TObject, TKey> _pivotOn;
+        private readonly IObservable<Func<Node<TObject, TKey>, bool>> _predicateChanged;
 
-        public TreeBuilder([NotNull] IObservable<IChangeSet<TObject, TKey>> source, [NotNull] Func<TObject, TKey> pivotOn)
+        public TreeBuilder([NotNull] IObservable<IChangeSet<TObject, TKey>> source, [NotNull] Func<TObject, TKey> pivotOn, IObservable<Func<Node<TObject, TKey>, bool>> predicateChanged)
         {
             _source = source ?? throw new ArgumentNullException(nameof(source));
             _pivotOn = pivotOn ?? throw new ArgumentNullException(nameof(pivotOn));
+            _predicateChanged = predicateChanged ?? Observable.Return(DefaultPredicate);
         }
 
+        internal static Func<Node<TObject, TKey>, bool> DefaultPredicate => node => node.IsRoot;
```

Comment:
My reasoning for having it as `internal` was so that it could be referenced in unit tests. I think that having this as a field does make more sense, though.